From 88d56067f5af7a3725cb2cefc44f4efebca9e57d Mon Sep 17 00:00:00 2001
From: George Cherian <george.cherian@marvell.com>
Date: Tue, 30 Mar 2021 16:19:37 +0530
Subject: [PATCH 1338/1921] PCI: Avoid marking multifunction device in same
 iommu group

Update the Cavium ACS quirk so that multifunction devices are not added
to the same iommu group. Currently the CGX devices are assigned to
the same iommu group. Avoid such a situation by dropping the explicit
check on PCI_EXP_TYPE_ROOT_PORT.

Fixes: 2da0e66be5f9 (PCI: Apply Cavium ACS quirk to ThunderX2 and ThunderX3)

Change-Id: I49bd42ab12599a88896bc40b8f28c375c43c7547
Signed-off-by: George Cherian <george.cherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/48866
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/pci/quirks.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 2984ff2e1ea6..1e868d509b9e 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -4366,7 +4366,7 @@ static int pci_quirk_amd_sb_acs(struct pci_dev *dev, u16 acs_flags)
 
 static bool pci_quirk_cavium_acs_match(struct pci_dev *dev)
 {
-	if (!pci_is_pcie(dev) || pci_pcie_type(dev) != PCI_EXP_TYPE_ROOT_PORT)
+	if (!pci_is_pcie(dev))
 		return false;
 
 	switch (dev->device) {
-- 
2.31.1

